client/src/details/OSInfo.cpp

```diff
@@ -92,27 +67,28 @@ std::string GetLinuxMachineInfo()
     }
     return "Unknown";
 }
+
+#else
+
+std::string GetPlatform()

```

Comment:
I think we'd be better with a compile-time failure in this else block, e.g.

`#error Unsupported platform`